utility/src/main/java/org/oppia/util/parser/ImprovedBulletSpan.kt

```diff
@@ -0,0 +1,75 @@
+package org.oppia.util.parser
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Path
+import android.graphics.Path.Direction
+import android.text.Layout
+import android.text.Spanned
+import android.text.style.LeadingMarginSpan
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.util.R
+
+/**
+ * Copy of [android.text.style.BulletSpan] from android SDK 28 with removed internal code.
+ *
+ * Reference: https://github.com/davidbilik/bullet-span-sample
+ */
+class ImprovedBulletSpan(activity: AppCompatActivity) : LeadingMarginSpan {
+
+  private var bulletRadius: Int = 0
+  private var gapWidth: Int = 0
+  private var yOffset: Int = 0
+
+  init {
+    bulletRadius = (activity as Context).resources.getDimensionPixelSize(R.dimen.bullet_radius)
```

Comment:
Why is the ``activity as Context`` bit necessary?

Reply:
This has been replaced with `activity` which is needed to access dimensions from `dimens.xml`

Reply:
Update: Removed activity entirely and directly used context here, which we are getting from `htmlParserTextView`